assets/js/modules/analytics/components/module/ModulePopularPagesWidget/index.js

```diff
@@ -72,20 +72,27 @@ export default function ModulePopularPagesWidget( { Widget, WidgetReportError }
 	const loaded = useSelect( ( select ) => select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [ args ] ) );
 	const error = useSelect( ( select ) => select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ args ] ) );
 
-	if ( error ) {
-		return <WidgetReportError error={ error } />;
+	if ( ! loaded ) {
+		// Return NULL while loading to replicate the behavior of the legacy components.
+		return null;
```

Comment:
While this in principle makes sense to exactly replicate the old behavior, let's not do this here. Let's build out each widget component individually, unaware of its surroundings, the same way the refactored version of the dashboard is built. In other words, this shouldn't return `null` just because we assume the `ModuleOverviewWidget` already takes care of the loading state.

We can later combine loading states etc as an enhancement if needed. I'd say it's more about consistency among all our widgets here - i.e. this should use `PreviewBlock`, wrapped within the `Widget` UI below.

Reply:
Updated.